src/Model.php

```diff
@@ -1528,28 +1517,17 @@ public function asModel($class, $options = [])
      * Create new model from the same base class
      * as $this.
      *
-     * @param string|Model $class
-     * @param array        $options
-     *
-     * @return Model
+     * @return static
      */
-    public function newInstance($class = null, $options = [])
+    public function newInstance(string $class = null, array $options = [])
```

Comment:
this will break compatibility. not sure if that's ok.

Reply:
That is correct, but to put this project in good shape we need it - as far as I remember these functions were used in a very few places.